### PR TITLE
Jest Tests

### PR DESCRIPTION
**Pull request by fed1k**
# Hi Code Reviewer. We ( @oshanedesign  and @fed1k ) followed the requirements below:
   * Write snapshot tests for all components
   * Write tests for `calculate.js` and `operate.js` files
## Happy Reviewing 🐱‍👤

### CODE REVIEW

File: src/logic/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import operate from './operate';
+// import { test, expect } from '@testing-library/jest-dom'
+
+it('should be one of the operators and should not be string', () => {
+  expect(operate('1', '4', '-')).toMatch(/.+/i);
+  expect(operate('8', '4', '-')).not.toMatch(/a-zA-Z/i);
+});
```
Comment (keroloussamy) [inline src/logic/operate.test.js]: - [x] Kindly add to your `operate.test.js` some tests for addition, multiple, modules, division, and subtraction. 

For example:
`expect(operate(10, 2, '÷')).toBe('5');`

File: src/logic/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    next: '0',
+  };
+
+  const obj = {
+    total: null,
+    next: '0',
+    operation: null,
+  };
+  expect(calculate(object, 'AC')).toStrictEqual(obj);
+});
```
Comment (keroloussamy) [inline src/logic/calculate.test.js]: - [x] Kindly add to your `calculate.test.js` some tests for addition, multiple, modules, division, subtraction, and some signs like `.`, `=`.

For example:
```  
test('calculate  6÷2=3', () => {
    expect(calculate({ total: '6', next: '2', operation: '÷' }, '=')).toStrictEqual({ total: '3', next: null, operation: null });
  });
```

File: src/components/Calculator.test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Design from './Calculator';
+
```
Comment (keroloussamy) [inline src/components/Calculator.test.js:5]: - [x] Kindly as the requirements say `Use React Testing Library to simulate user interaction.` Kindly add some tests for user interactions with the calculator.

For example
```
test('Test Calculator', async () => {
  render(<Calculator />);
  fireEvent.click(screen.getByText('9'));
  const result = await waitFor(() => screen.getByTestId('result'));
  expect(result.textContent).toBe('9');
});
```

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._

- [x] Kindly make your partner add commits equally to your commits also, so the work will be divided equally among you.



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @fed1k, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**